src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerAttributeBindingProvider.cs

```diff
@@ -20,13 +22,16 @@ internal class CosmosDBTriggerAttributeBindingProvider : ITriggerBindingProvider
         private readonly INameResolver _nameResolver;
         private string _monitorConnectionString;
         private string _leasesConnectionString;
+        private TraceWriter _trace;
         private CosmosDBConfiguration _config;
+        private readonly ILogger _logger;
 
-        public CosmosDBTriggerAttributeBindingProvider(INameResolver nameResolver, CosmosDBConfiguration config, ChangeFeedHostOptions leasesOptions = null)
+        public CosmosDBTriggerAttributeBindingProvider(INameResolver nameResolver, CosmosDBConfiguration config, ILoggerFactory loggerFactory, ChangeFeedHostOptions leasesOptions = null)
         {
             _nameResolver = nameResolver;
             _config = config;
             _leasesOptions = leasesOptions ?? new ChangeFeedHostOptions();
+            _logger = loggerFactory?.CreateLogger(LogCategories.CreateTriggerCategory("CosmosDB"));
```

Comment:
Why would this ever be null? I was thinking you'd just require non-null here then all the way down logger would be non-null. Seems easier than having to check for a null logger each time you use it.